pkg/cloud/services/networking/port.go

```diff
@@ -347,11 +347,9 @@ func (s *Service) CreatePorts(eventObject runtime.Object, desiredPorts []infrav1
 }
 
 // ConstructPorts builds an array of ports from the machine spec.
-// If no ports are in the spec, returns a single port for a network connection to the default cluster network.
-func (s *Service) ConstructPorts(spec *infrav1.OpenStackMachineSpec, clusterResourceName, baseName string, defaultNetwork *infrav1.NetworkStatusWithSubnets, managedSecurityGroup *string, baseTags []string) ([]infrav1.ResolvedPortSpec, error) {
-	ports := spec.Ports
-
-	defaultSecurityGroupIDs, err := s.GetSecurityGroups(spec.SecurityGroups)
+// If no ports are in the spec, returns an error.
+func (s *Service) ConstructPorts(portsOpts []infrav1.PortOpts, securityGroups []infrav1.SecurityGroupParam, instanceTrunk bool, clusterResourceName, baseName string, managedSecurityGroup *string, baseTags []string) ([]infrav1.ResolvedPortSpec, error) {
```

Comment:
for myself, check whether `securityGroups` could be renamed to `defaultSecurityGroups`?